src/reducers/entities/threads.ts

```diff
@@ -87,6 +87,40 @@ export const threadsInTeamReducer = (state: ThreadsState['threadsInTeam'] = {},
 };
 export const countsReducer = (state: ThreadsState['counts'] = {}, action: GenericAction) => {
     switch (action.type) {
+    case ThreadTypes.ALL_TEAM_THREADS_READ: {
+        const counts = state[action.data.team_id] ?? {unread_mentions_per_channel: {}};
+        return {
+            ...state,
+            [action.data.team_id]: {
+                ...counts,
+                unread_mentions_per_channel: {},
+                total_unread_mentions: 0,
+            },
+        };
+    }
+    case ThreadTypes.READ_CHANGED_THREAD: {
```

Comment:
@calebroseland - let me elaborate on the reason for 'diff':
We have unread_mentions_per_channel that counts unread mentions in all threads by channel.
When a thread is marked as unread - the count for the channel has to go up
When a thread is marked as read - the count for the channel has to go down
So the diff let's me know how to update the per-channel count.
In order to avoid sync/network issues, I'm going to rewrite the backend part to return the 'new mentions' and update webapp to calculate the diff by itself.
Sounds good?